app/src/main/java/org/openobservatory/ooniprobe/model/database/Result.java

```diff
@@ -94,9 +94,21 @@ public long countOkMeasurements() {
 	public long countAnomalousMeasurements() {
 		return SQLite.selectCountOf().from(Measurement.class).where(Measurement_Table.result_id.eq(id), Measurement_Table.is_rerun.eq(false), Measurement_Table.is_done.eq(true), Measurement_Table.is_failed.eq(false), Measurement_Table.is_anomaly.eq(true)).count();
 	}
+	
+	private Measurement getFirstMeasurement() {
+		return SQLite.select().from(Measurement.class).where(Measurement_Table.result_id.eq(id), Measurement_Table.is_rerun.eq(false)).orderBy(Measurement_Table.start_time, true).querySingle();
+	}
+
+	private Measurement getLastMeasurement() {
+		return SQLite.select().from(Measurement.class).where(Measurement_Table.result_id.eq(id), Measurement_Table.is_rerun.eq(false)).orderBy(Measurement_Table.start_time, false).querySingle();
+	}
 
-	public void addDuration(double value) {
-		this.runtime += value;
+	public double getRuntime(){
+		Measurement first = getFirstMeasurement();
+		Measurement last = getLastMeasurement();
```

Comment:
Is there a possibility of these not being defined? What happens if there are no rows for the specified search criteria can do we get an exception or maybe an undefined model?

Can we add some extra checks for this?

Reply:
You can't get nil because you can't access the test summary screen if there are no measurement in the result. See https://github.com/ooni/probe/issues/855